### PR TITLE
Added many features to Entity, Ped, Vehicle, World, UI

### PR DESCRIPTION
**Pull request by kagikn**

### CODE REVIEW

**Comment by crosire** [comment]
Are UI::Width and UI::Height now returning the actual screen resolution? Because that would break the entire menu system compared to how it worked until now (with absolute distances based on a fixed resolution). Not saying that is bad, but merging that now would break pretty much every script with a menu.

**Comment by kagikn** [comment]
Yeah, I confirmed _GET_SCREEN_ACTIVE_RESOLUTION returns actual screen resolution.

**Comment by JohnnyCrazy** [comment]
Then we also need to think of a new UI system, don't we? Else everybody with a different screen resoultion will have a lot of trouble.

File: source/Entity.cpp [hunk 1]
```diff
@@ -223,6 +223,23 @@ namespace GTA
 		Native::Function::Call(Native::Hash::APPLY_FORCE_TO_ENTITY, this->Handle, 3, direction.X, direction.Y, direction.Z, rotation.X, rotation.Y, rotation.Z, false, true, true, true, false, true);
 	}
 
+	unsigned char Entity::GetAlpha()
```
Comment (zorg93) [inline source/Entity.cpp]: Why is this an unsigned char, does that show as a byte in c#

**Comment by frodzet** [comment]
@JohnnyCrazy Why use int? If it ranges from 0 to 255, would it matter anyways?

**Comment by JohnnyCrazy** [comment]
2 reasons:

1. This simple code wouldn't work, while it has to for simple arithmetic stuff
img

2. [Quote from stackoverflow](http://stackoverflow.com/a/2346398/1437049)
>Usually yes, a 32 bit integer will perform slightly better because it is already properly aligned for native CPU instructions. You should only use a smaller sized numeric type when you actually need to store something of that size.

But you can pretty much ignore the second argument and stick to the first :P

**Comment by frodzet** [comment]
@JohnnyCrazy Thanks for clarifying :-)

**Pull request merged**